src/wallet/wallet.cpp

```diff
@@ -86,8 +86,10 @@ std::string COutput::ToString() const
 
 int COutput::Priority() const
 {
-    for (const auto& d : CPrivateSend::GetStandardDenominations())
-        if(tx->tx->vout[i].nValue == d) return 10000;
+    for (const auto& d : CPrivateSend::GetStandardDenominations()) {
+        // large denoms have lower value
+        if(tx->tx->vout[i].nValue == d) return (float)COIN / d * 10000;
```

Comment:
I'm not very familiar with c++ type casting, is this `return (float)...` ok here?

Reply:
tl;dr: should be ok.

First, `COIN` (`int64_t`) is casted into `float` so that we won't end up with `0` when `COIN < d`. Then, the result is multiplied by `10000` (and becomes `>1`) but it's still a `float` which is casted to `int` (the type returned by the function) on return automagically.